Framework/PythonInterface/plugins/algorithms/LoadWANDSCD.py

```diff
@@ -48,46 +117,104 @@ def validateInputs(self):
         return issues
 
     def PyExec(self):
-        runs = self.getProperty("Filename").value
-
-        if not runs:
+        # Process input data
+        # get the list of input filenames
+        runs = self.get_intput_filenames()
+        # load and group
+        data = self.load_and_group(runs)
+
+        # Process the vanadium normalization data
+        if self.getProperty("NormalizedBy").value.lower() == 'none':
+            # no normalization
+            self.getLogger().information("Skip Va normalization as requested")
+        else:
+            van_filename = self.get_va_filename()
+            if van_filename is None:
+                # try to load from memory
+                norm = self.getProperty("VanadiumWorkspace").value
+            else:
+                norm = self.load_and_group([van_filename])
+            # normalize the results
+            data = self.normalize(data, norm, self.getProperty("NormalizedBy").value.lower())
+            # cleanup
+            DeleteWorkspace(norm)
```

Comment:
You should probably only delete this if you loaded it, so don't delete `norm` if it came from `VanadiumWorkspace`. This would also allow you to not `reload` the vanadium 3 times in the test.

Reply:
that's right, I will make the change now